src/channeldef.ts

```diff
@@ -198,6 +199,72 @@ export type TextDef<F extends Field> =
   | FieldOrDatumDefWithCondition<StringDatumDef<F>, Text>
   | ValueDefWithCondition<StringFieldDef<F>, Text>;
 
+/**
+ * Position of a label relative to its base mark's position, which includes anchor direction (anchor) and distance (offset).
+ */
+export type LabelPosition = {
+  /**
+   * distance of a label from its base mark.
+   * if negative, the label position is insize.
+   */
+  offset: number;
+
+  /**
+   * anchor direction of a label from its base mark.
+   * The value could be one of `'left '` , `'right '` , `'top '` , `'middle '` ,
+   * `'bottom '` , `'top-left '` , `'top-right '` , `'bottom-left '` , or  '`bottom-right '`
+   */
+  anchor: LabelAnchor;
+};
+
+export type LabelDefMixins = {
+  /**
+   * A list of possible positions for a label to be placed relative to its base mark.
+   */
+  position?: LabelPosition[];
+
+  /**
+   * The layers of marks that the labels will avoid.
+   * `'all'` means all the marks that are children of the root layer.
+   * number `n` means all the marks that are children of the nth ancestor layer.
+   *
+   * __Default value:__ `0` (only avoid the base mark)
+   */
+  avoidParentLayer?: 'all' | number;
+
+  /**
+   * Mark definition of the label text mark without the `'type'` property.
```

Comment:
Just `type` in fine.  No need to do quotes for property names.

Reply:
Actually

```
Mark definition of the label text mark without the `'type'` property.
```

describes what the type looks like, not what the type is for. 

I think it should be something like:

```
A label mark definition for customizing the label's text mark. 
This definition includes all [properties of a text mark](http://vega.github.io/vega-lite/docs/text.html#properties) except `type`. 
```